tests/state_test_util.go

```diff
@@ -144,15 +144,17 @@ type stAuthorization struct {
 	Address common.Address `json:"address" gencodec:"required"`
 	Nonce   uint64         `json:"nonce" gencodec:"required"`
 	V       uint8          `json:"v" gencodec:"required"`
-	R       uint256.Int    `json:"r" gencodec:"required"`
-	S       uint256.Int    `json:"s" gencodec:"required"`
+	R       *big.Int       `json:"r" gencodec:"required"`
+	S       *big.Int       `json:"s" gencodec:"required"`
 }
 
 // field type overrides for gencodec
 type stAuthorizationMarshaling struct {
 	ChainID math.HexOrDecimal64
 	Nonce   math.HexOrDecimal64
 	V       math.HexOrDecimal64
+	R       *math.HexOrDecimal256
+	S       *math.HexOrDecimal256
```

Comment:
AFAIK, the default unmarshaler for uint256 accepts hex or decimal. It uses the prefix `0x` to detect hex.
https://github.com/holiman/uint256/blob/a17fcfb6f8b0245533928fa96aace2cd4c2e9b18/conversion.go#L236-L239

Reply:
The issue with `uint256` is it does not accept leading zeros: https://github.com/holiman/uint256/blob/a17fcfb6f8b0245533928fa96aace2cd4c2e9b18/conversion.go#L844-L846

I assumed we'd want to leave the behavior there as-is, but if you prefer we could change it there instead?

Reply:
Maybe the question is, do they even use decimal or hex with leading zero in tests? The only reason why we support a more relaxed encoding in state tests, is because tests used to be written by hand in the old days, and their encoding was not always consistent. I would assume the tests have a consistent format now?

Reply:
Yeah they encode integers to always have even length, which means `1` is encoded `0x01`.